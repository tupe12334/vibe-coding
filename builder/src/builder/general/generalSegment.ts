import { readFile } from "fs/promises";
import { join } from "path";

export const generalSegment = async (templatesPath: string) => {
  const generalJsonFile = (
    await readFile(join(templatesPath, "general.json"), {
      encoding: "utf-8",
    })
  ).toString();
  const generalObject = JSON.parse(generalJsonFile);

  const generalSegment = [
    {
      type: "heading",
      depth: 2,
      children: [{ type: "text", value: "General Guidelines" }],
    },
    { type: "heading", depth: 3, children: [{ type: "text", value: "Do" }] },
    ...generalObject.do.map((item: string) => ({
      type: "listItem",
      children: [{ type: "text", value: item }],
    })),
  ];
  return generalSegment;
};
