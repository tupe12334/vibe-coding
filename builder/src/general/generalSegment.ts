import { readFile } from "fs/promises";
import { join } from "path";

import { RootContent } from "mdast";

export const generalSegment = async (
  templatesPath: string
): Promise<RootContent[]> => {
  const generalJsonFile = (
    await readFile(join(templatesPath, "general.json"), {
      encoding: "utf-8",
    })
  ).toString();
  const generalItems = JSON.parse(generalJsonFile);

  const generalSegment: RootContent[] = [
    {
      type: "heading",
      depth: 2,
      children: [{ type: "text", value: "General Guidelines" }],
    },
    ...generalItems.map((item: string) => ({
      type: "listItem",
      children: [{ type: "text", value: item }],
    })),
  ];
  return generalSegment;
};
