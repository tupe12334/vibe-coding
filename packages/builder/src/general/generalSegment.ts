import type { ListItem, RootContent } from "mdast";
import { generalRules } from "./rules";

export const generalSegment = async (): Promise<RootContent[]> => {
  const generalItems = generalRules;

  const segment: RootContent[] = [
    {
      type: "heading",
      depth: 2,
      children: [{ type: "text", value: "General Guidelines" }],
    },
    ...generalItems.map(
      (item): ListItem => ({
        type: "listItem",
        children: [
          { type: "paragraph", children: [{ type: "text", value: item }] },
        ],
      })
    ),
  ];
  return segment;
};
