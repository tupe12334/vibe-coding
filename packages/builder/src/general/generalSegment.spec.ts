import { describe, expect, test } from "vitest";
import { generalSegment } from "./generalSegment";

describe("generalSegment", () => {
  test("returns general guidelines", async () => {
    const segment = await generalSegment();
    expect(segment).toMatchSnapshot();
  });
});
