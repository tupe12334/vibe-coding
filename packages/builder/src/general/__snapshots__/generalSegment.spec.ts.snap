// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generalSegment > returns general guidelines 1`] = `
[
  {
    "children": [
      {
        "type": "text",
        "value": "General Guidelines",
      },
    ],
    "depth": 2,
    "type": "heading",
  },
  {
    "children": [
      {
        "children": [
          {
            "type": "text",
            "value": "Don't work on any of the tasks in the TODO.md file unless you are asked to.",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "listItem",
  },
  {
    "children": [
      {
        "children": [
          {
            "type": "text",
            "value": "Always try to early return from functions.",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "listItem",
  },
  {
    "children": [
      {
        "children": [
          {
            "type": "text",
            "value": "Use \`const\` for variables that are not reassigned.",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "listItem",
  },
  {
    "children": [
      {
        "children": [
          {
            "type": "text",
            "value": "Make sure to focus on why and not how in documentation.",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "listItem",
  },
  {
    "children": [
      {
        "children": [
          {
            "type": "text",
            "value": "Write predictable functions and make a spec file for them according to the testing library in use.",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "listItem",
  },
  {
    "children": [
      {
        "children": [
          {
            "type": "text",
            "value": "Don't use \`any\` in TypeScript, use \`unknown\` instead.",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "listItem",
  },
  {
    "children": [
      {
        "children": [
          {
            "type": "text",
            "value": "Don't cast types without validation.",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "listItem",
  },
  {
    "children": [
      {
        "children": [
          {
            "type": "text",
            "value": "Whenever finish a task from the TODO.md file mark it as finished, if there is some that already marked as finish delete them",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "listItem",
  },
  {
    "children": [
      {
        "children": [
          {
            "type": "text",
            "value": "Work in a domain driven design (DDD) way, i.e. every function its is own module, and its its folder there is the code with JSDocs, spec file and types in there files.",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "listItem",
  },
]
`;
